sqldb/migrations.go

```diff
@@ -381,6 +387,30 @@ func ApplyMigrations(ctx context.Context, db *BaseDB,
 			"(dirty=%v) as base version", currentVersion, dirty)
 	}
 
+	// Due to an a migration issue in v0.19.0-rc1 we may be at version 2 and
+	// have a dirty schema due to failing migration 3. If this is indeed the
+	// case, we need to reset the dirty flag to be able to apply the fixed
+	// migration.
+	// NOTE: this could be removed as soon as we drop v0.19.0-beta.
+	if version == 2 {
+		schemaVersion, dirty, err := migrator.GetSchemaVersion()
+		if err != nil {
+			return err
+		}
+
+		if schemaVersion == 3 && dirty {
```

Comment:
something im not getting:

afaiu, if `dirty==true`, then the `schemaVersion` was actually _not_ successfully applied. So if we are changing `dirty=false` then should we not then set `schemaVersion=2` so that migration 3 gets properly retried?

i understand that the edge case we are catching here is for the unique case where actually migration 3 already ran in the past and we are now re-running it accidentally in `19.rc1` - but just wondering if there is not perhaps a case where migration 3 is actually being run for the first time and fails for some different reason

Reply:
You're right, technically we should go back to to schema version 2, in this specific case i think we're safe to jump to 3 or even 4 as we know the user was already there.